inc/namespace.php

```diff
@@ -33,67 +33,148 @@ function register() {
  * Bootstrap module, when enabled.
  */
 function bootstrap() {
+	add_filter( 'altis.documentation.sets', __NAMESPACE__ . '\filter_add_dev_docs_set' );
+	add_filter( 'altis.documentation.default.group', __NAMESPACE__ . '\get_default_group', 10, 2 );
 	UI\bootstrap();
 }
 
 /**
- * Get all documentation groups.
+ * Get the required documentation set.
+ * Returns an empty set if not found.
  *
- * @return Group[] Sorted list of groupse
+ * @param string $set_id The required set id.
+ *
+ * @return Set The required Documentation Set or a new empty one.
  */
-function get_documentation() : array {
-	static $docs;
+function get_documentation_set( string $set_id ) : Set {
+	/**
+	 * All the documentation sets.
+	 *
+	 * @var Set[] $all_sets All the documentation sets.
+	 */
+	static $all_sets = [];
 
-	if ( ! empty( $docs ) ) {
-		return $docs;
+	/**
+	 * Filter available documentation sets.
+	 *
+	 * This allows modules to register additional documentation sets or replace the requested set.
+	 *
+	 * @param Set[] $doc_sets Map of set ID to Set object.
+	 * @param string $set_id The required set id.
+	 */
+	$all_sets = apply_filters( 'altis.documentation.sets', $all_sets );
+
+	if ( ! empty( $all_sets[ $set_id ] ) ) {
+		return $all_sets[ $set_id ];
 	}
 
-	$modules = Module::get_all();
+	return new Set();
+}
+
+/**
+ * Filter the $all_sets array to add the developer docs Set if it doesn't yet exist.
+ *
+ * @param array $sets The array of Documentation sets.
+ *
+ * @return array
+ */
+function filter_add_dev_docs_set( array $sets ) : array {
+
+	// Are we already set up?
+	if ( ! empty( $sets['dev-docs'] ) ) {
+		return $sets['dev-docs'];
```

Comment:
This would fatal as `$sets['dev-docs']` would be a `Set` and not an `array`

Reply:
Doh! Fixed.